Move training to Modal

corbt opened the pull request:
Use Modal for training and inference.

corbt committed:
initial training script import

corbt committed:
Trainer script

Able to pull data from the db and train a model based on it. Pushes the model to HF. Unsure if the model works.

corbt committed:
deploy to baseten

corbt committed:
Move training to Modal

Use Modal for training and inference.

arcticfly commented on .tool-versions:
```diff
@@ -1 +1,2 @@
 nodejs 20.2.0
+python 3.9.13
```
Nit: add newline?

arcticfly commented on app/Dockerfile:
```diff
@@ -29,7 +39,7 @@ COPY --from=deps /code/node_modules ./node_modules
 COPY --from=deps /code/client-libs/typescript/node_modules ./client-libs/typescript/node_modules
 COPY . .
-RUN cd app && SKIP_ENV_VALIDATION=1 pnpm build
+# RUN cd app && SKIP_ENV_VALIDATION=1 pnpm build
```
Why did you comment out the build step?

corbt replied:
Accidentally committed this, was just to make local testing faster. Added back.

arcticfly commented on app/prisma/schema.prisma:
```diff
@@ -189,15 +188,15 @@ enum DatasetFileUploadStatus {
-    status   DatasetFileUploadStatus @default(PENDING)
-    uploadedAt DateTime
-    visible  Boolean @default(true)
+    datasetId    String                  @db.Uuid
```
Do you have a formatter for prisma files?

corbt replied:
Yeah just the one built into their vs code [extension](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma)

arcticfly commented on app/scripts/copy-db.sh:
Why did you remove the production debugging script?

corbt replied:
Moved it, didn't remove.

arcticfly commented on app/src/modelProviders/fine-tuned/getCompletion-2.ts:
```diff
@@ -0,0 +1,114 @@
+    throw new Error("Streaming is not yet supported");
+  }
+
+  const resp = await fetch(`https://openpipe--${modalDeployId}.modal.run`, {
```
I suppose we can get rid of inference urls sometime soon.

arcticfly commented on app/src/modelProviders/fine-tuned/getCompletion-2.ts:
```diff
@@ -0,0 +1,114 @@
+  };
+}
+
+export const formatInputMessages = (
```
Is this all copy/pasted from getCompletion? I would prefer to define them in one place and import from there.

arcticfly commented on app/src/server/tasks/fineTuning/deployFineTune.task.ts:
```diff
@@ -0,0 +1,83 @@
+  const modalDeployId = `ft-${env.NODE_ENV}-${fineTuneId}`;
+
+  // Inline setupTemporaryEnvironment
+  const tempDir = await fs.mkdtemp(path.join(os.tmpdir(), "deploy-"));
```
Should we incorporate the deploy id into the tempDir name?

corbt replied:
Now outdated

arcticfly commented on app/src/server/tasks/fineTuning/deployFineTune.test.ts:
```diff
@@ -0,0 +1,34 @@
+    await deployFineTune(ft.id);
+  },
+  {
+    timeout: 1000 * 60 * 60 * 20,
```
20 hour timeout, I like it

arcticfly replied:
We need to make sure to add this to our CI/CD 😆

arcticfly commented on app/src/tests/helpers/globalSetup.ts:
```diff
@@ -0,0 +1,10 @@
+import "./loadEnv";
+import { $ } from "execa";
+
+// await $({ stdio: "inherit" })`pnpm prisma migrate reset --force`;
```
Do we need this commented-out line?

corbt replied:
Removed

arcticfly commented on trainer/pyproject.toml:
```diff
@@ -2,23 +2,18 @@
 name = "trainer"
 version = "0.1.0"
 description = ""
-authors = ["David Corbitt <davidlcorbitt@gmail.com>"]
```
Shots fired

corbt replied:
Lol sorry that actually wasn't intentional, at one point in this PR I was moving away from poetry and onto pip entirely so I deleted all the poetry-generated config and then ended up adding it back in. Added back. 😂

arcticfly commented on trainer/trainer/entrypoint.py:
```diff
@@ -0,0 +1,40 @@
+    do_train(fine_tune_id, base_url, api_key)
+
+
+@stub.local_entrypoint()
```
Does this only get called when we're running the server on our own machines? In the production environment, what calls `train`?

corbt replied:
Now outdated.

arcticfly reviewed (approved):
Nice work! I'm very excited to try this out.

corbt committed:
More Modal work

Make it so we have just one inference and one trainer app. Don't try to deploy a new app for each model.

corbt committed:
minor fixes

corbt committed:
Merge remote-tracking branch 'origin/main' into train-script
